### ISSUE TITLE
support icon_url field

### ISSUE THREAD

**Issue by YoungjaeKim** [alexkvak/teamcity-slack#49]
First of all, thanks for the great plugin.

#15 reported that icon is displayed in default bot icon, and so do I. 
Even though I properly set bot profile image, Teamcity-sent message is shown in default bot icon.

Could you provide `icon_url` field for the message?
- https://api.slack.com/methods/chat.postMessage

**Comment by alexkvak** [comment]
So you would like to specify icon url on global plugin settings page?

**Comment by YoungjaeKim** [comment]
@alexkvak thanks for the reply.
Yes, correct.

I've already customized bot profile image at bot settings, but the message from this plugin only shows _default_ bot icon.

The message from this plugin is as below;
![image](https://user-images.githubusercontent.com/1265284/46662986-d1adf780-cbf7-11e8-9d94-074d423c361c.png)

But when I click the profile, the bot profile image is already defined as below;
![image](https://user-images.githubusercontent.com/1265284/46663006-e4c0c780-cbf7-11e8-89db-c2f31e3a1d72.png)

So, I thought that this issue could be resolved by the plugin global settings page, and the plugin consumes it by using `icon_url` Slack api field.

**Comment by YoungjaeKim** [comment]
I'm sorry it's quite confused; what is `Slack TeamCity user` icon?
There's no `TeamCity` _user account_ actually, as this account is created by adding bot app.

I've tried two bot apps. both are providing API Token.
1. Bots; https://clova.slack.com/apps/A0F7YS25R-bots?next_id=0
2. Hubot; https://clova.slack.com/apps/A0F7XDU93-hubot

Note that my organization restricts bot apps and Hubot is the only one that I can use now.

**Comment by alexkvak** [comment]
> Note that my organization restricts bot apps and Hubot is the only one that I can use now.
Probably this is the reason you don't have an icon from app?

Please also check whether your app is added to the channel.

**Comment by YoungjaeKim** [comment]
@alexkvak Yes, the bot is added to channels. That's why the bot can send messages to the channel by your plugin :) The bot configuration page shows its invited channels and I confirmed that it seems OK.

**Comment by YoungjaeKim** [comment]
@alexkvak I found that your bot settings page and mine look slightly different. Mine is as below;

![image](https://user-images.githubusercontent.com/1265284/46835303-b9e99580-cde8-11e8-90c8-5c089760588c.png)

According to [document](https://api.slack.com/methods/chat.postMessage) *Authorship* section, I think the _bot_ we are saying could be different type; Slack App user token, and Slack App bot user token? 
Or, mine is probably a [workspace app](https://api.slack.com/bot-users#getting-started-workspace).

Whatever it is, it seems they could work differently. (I don't believe the document freshness, but we can check that there can be _differences_ between two).

I assume that `icon_url` on each message could resolve the profile photo issue as this field will guarantee an icon for the one message.

**Comment by LazyTarget** [comment]
I also am in need for this feature.
My Slack configuration looks like @YoungjaeKim except instead of "Hubot" it simply says "Bots". 
With the url: _https://{TEAM}.slack.com/apps/A0F7YS25R-bots_

I believe, as @YoungjaeKim originally mentioned that the `icon_url` field needs to be set, using a global configuration for the Slack plugin (as like with the SenderName)

> Could you provide `icon_url` field for the message?
> 
> * https://api.slack.com/methods/chat.postMessage

I'd be happy to implement this although.. my knowledge of scala is basically zero
